pkg/skaffold/initializer/init.go

```diff
@@ -40,64 +41,99 @@ func DoInit(ctx context.Context, out io.Writer, c config.Config) error {
 		}
 	}
 
+	a, err := AnalyzeProject(c)
+	if err != nil {
+		return err
+	}
+
+	newConfig, newManifests, err := Initialize(out, c, a)
+	if err != nil || c.Analyze {
+		return err
+	}
+
+	return WriteData(out, c, newConfig, newManifests)
+}
+
+// AnalyzeProject scans the project directory for files and keeps track of what types of files it finds (builders, k8s manifests, etc.).
+func AnalyzeProject(c config.Config) (*analyze.ProjectAnalysis, error) {
 	a := analyze.NewAnalyzer(c)
 	if err := a.Analyze("."); err != nil {
-		return err
+		return nil, err
 	}
 
 	// helm projects can't currently be bootstrapped automatically by skaffold, so we fail fast and link to our docs instead.
 	if len(a.ChartPaths()) > 0 {
 		//nolint
-		return errors.New(`Projects set up to deploy with helm must be manually configured.
+		return nil, errors.New(`Projects set up to deploy with helm must be manually configured.
 
 See https://skaffold.dev/docs/pipeline-stages/deployers/helm/ for a detailed guide on setting your project up with skaffold.`)
 	}
 
+	return a, nil
+}
+
+// Initialize uses the information gathered by the analyzer to create a skaffold config and generate kubernetes manifests.
```

Comment:
I see 3 return type. 
Can you add more description on what the `map` represents and when it return an error ?

Reply:
I added a comment to explain the the map that returns. Would you like me to explain each error scenario?